specs/interop/shared-locbox.md

```diff
@@ -0,0 +1,300 @@
+It assumes the role of a `dependencyManager` per `SystemConfig` contract involved.
+It will also allow to add a chain to the op-governed cluster and update each chain’s dependency set.
+
+**Storage layout**
+
+The `SuperchainConfig` contract will contain the following storage layout:
+
+1. `SHARED_LOCKBOX`: An immutable address pointing to the Shared Lockbox contract.
+   This address is immutable because there's only one `SharedLockbox` for each cluster.
+2. `systemConfigs`: A mapping that associates chain IDs with their respective SystemConfig addresses.
+   This will be used when updating dependencies along each chain.
+3. `_dependencySet`: An EnumerableSet that stores the current list of chain IDs in the dependency set.
+   This should replicate the same state as the one stored in the `L1BlockInterop` on L2.
+
+```solidity
+// The Shared Lockbox contract
+address internal immutable SHARED_LOCKBOX;
+
+// Mapping from chainId to SystemConfig address
+mapping(uint256 _chainId => ISystemConfig) public systemConfigs;
+
+// Current dependency set list
+EnumerableSet.UintSet internal _dependencySet;
+```
+
+**`addChain`**
+
+The `addChain` function adds a new chain to the op-governed cluster.
+This function can only be called by the `updater`.
+It tries to add the chain’s ID to the dependency set, making sure it’s not already included.
+The function stores its system config and updates dependencies by linking the new chain with
+all existing chains in the set, making a full mesh graph.
+It finally adds the portal to the `ShareLockbox` whitelist.
+
+```solidity
+function addChain(uint256 _chainId, address _systemConfig) external {
+    require(msg.sender == updater(), "Unauthorized");
+
+    // Add to the dependency set and check it is not already added (`add()` returns false if it already exists)
+    require(_dependencySet.add(_chainId), "Chain already added");
+
+    // Store the system config
+    systemConfigs[_chainId] = _systemConfig;
+
+    // Loop through the dependency set and update the dependency for each chain
+    for (uint256 i; i < _dependencySet.length() - 1; i++) {
+        uint256 currentId = _dependencySet.at(i);
+
+        // Skip recently added chain
+        if (_chainId == currentId) continue;
+
+        systemConfigs[currentId].addDependency(_chainId);
+
+        systemConfigs[_chainId].addDependency(currentId);
+    }
+
+    address portal = _systemConfig.optimismPortal();
+
+    // Authorize the portal on the shared lockbox
+    SHARED_LOCKBOX.authorizePortal(portal);
+
+    emit ChainAdded(_chainId, _systemConfig, portal);
+}
+```
+
+**Getters**
+
+These getter are needed due to using an EnumerableSet.
+
+```solidity
+function dependencySet() external view returns (uint256[] memory) {
+    return dependencySet.values();
+}
+
+function isInDependencySet(uint256 _chainId) public view returns (bool) {
+    return dependencySet.contains(_chainId);
+}
+```
+
+### `OptimismPortal`
```

Comment:
Ok, so you can have a section called ## Upgrades
where you list these kind of things. Otherwise it seems like a new contract.